components/js/SlickGrid.ts

```diff
@@ -89,15 +54,32 @@ function getOverridableTextEditorClass(grid: SlickGrid): any {
         };
 
         applyValue(item, state): void {
-            this._textEditor.applyValue(item, state);
+            let currentRow = grid.dataRows.at(this._rowIndex);
+            let colIndex = grid.getColumnIndex(this._args.column.name);
+            let dataLength: number = grid.dataRows.getLength();
+
+            // If this is not the "new row" at the very bottom
+            if (this._rowIndex !== dataLength) {
+                currentRow.values[colIndex] = state;
+                this._textEditor.applyValue(item, state);
+            }
         };
 
         isValueChanged(): boolean {
             return this._textEditor.isValueChanged();
         };
 
         validate(): any {
-            return this._textEditor.validate();
+            let result: any =  { valid: true, msg: undefined };
+            let colIndex: number = grid.getColumnIndex(this._args.column.name);
+            let newValue: any = this._textEditor.getValue();
+
+            // TODO: It would be nice if we could support the isCellEditValid as a promise 
```

Comment:
It's likely this isn't possible (it's intended to complete in a short time). If we can't make it happen by wrapping our async call in a sync block, we need to look into alternative validation methods.